pkg/devfile/detect.go

```diff
@@ -43,102 +43,89 @@ type AlizerClient struct {
 // Map 1 returns a context to the devfile bytes if present.
 // Map 2 returns a context to the matched devfileURL from the devfile registry if no devfile is present in the context.
 // Map 3 returns a context to the dockerfile uri or a matched dockerfileURL from the devfile registry if no dockerfile is present in the context
-func search(log logr.Logger, a Alizer, localpath string, currentLevel, depth int, devfileRegistryURL string) (map[string][]byte, map[string]string, map[string]string, error) {
-	// TODO - maysunfaisal
-	// There seems to a gap in the logic if we extend past depth 1 and discovering devfile logic
-	// Revisit post M4
+func search(log logr.Logger, a Alizer, localpath string, devfileRegistryURL string) (map[string][]byte, map[string]string, map[string]string, error) {
```

Comment:
wondering what happens when in the future if they say we want to search for more than depth of 1. That was my initial idea to have it as a variable, so that only the API design will change and this function can perhaps handle searching deeper than 1 level

Reply:
IMO, since we are only looking for 1-levek subfolders, (and currently have no plan to expand deeper), I found it's hard to maintain a recursive function.  
and we can always get the only API design back, and well tested it if we want to search for more than 1 depth.